docs/pages/guides/monorepos.md

```diff
@@ -92,10 +93,80 @@ config.resolver.nodeModulesPaths = [
   path.resolve(projectRoot, 'node_modules'),
   path.resolve(workspaceRoot, 'node_modules'),
 ];
+// 3. Force Metro to resolve (sub)dependencies only from the `nodeModulesPaths`
+config.resolver.disableHierarchicalLookup = true;
 
 module.exports = config;
 ```
 
+<Collapsible summary="1. Why do we need to watch all files with the monorepo?">
+
+This option enables the app to refresh or reload when you edit imported files outside the app folder. It's important for libraries within your monorepo that are imported into your app. Without this setting, you must manually reload your app and possibly clear the Metro cache to see the changed library files.
+
+If your monorepo becomes really big, watching all files within the monorepo might be slow. If you want to speed that up, you can (dynamically) watch the relevant paths to minimize the watchers. E.g.
+
+```js
+const path = require('path');
+const workspaceRoot = path.resolve(__dirname, '../..');
+
+// If your monorepo tooling can give you a list of monorepo dependencies from a workspace,
+// you can automate this instead of hardcoding the paths.
+const appMonorepoPackages = ['ui', 'eslint-config'];
+const appPackagesPaths = appMonorepoPackages.map(
+  // Get the path of the package folder, not just the entry point.
+  packageName => path.resolve(workspaceRoot, 'packages', packageName)
+);
+
+// 1. Watch the local app folder, and all related packages (limiting the watchers and speeding it up)
+config.watchFolders = [__dirname, ...appPackagesPaths];
+```
+
+</Collapsible>
+
+<Collapsible summary="2. Why do we need tell Metro how to resolve packages?">
+
+This option is important to resolve libraries in the right **node_modules** folders. Monorepo tooling, like Yarn, usually creates two different **node_modules** folders which are used for a single workspace.
```

Comment:
```suggestion
This option is important to resolve libraries in the correct **node_modules** directories. Monorepo tooling, like Yarn, usually creates two different **node_modules** directories which are used for a single workspace.
```

not so picky about it but i generally prefer directory over folder. what do you think? if you agree, maybe we should enforce this automatically? i'd at least like to be intentional about which term we use and apply it consistently everywhere

Reply:
I don't care too much either, I usually use folder because it's shorter.